packages/mdc-top-app-bar/component.ts

```diff
@@ -57,19 +61,52 @@ export class MDCTopAppBar extends MDCComponent<MDCTopAppBarBaseFoundation> {
     this.scrollTarget_ = window;
   }
 
+  initialSyncWithDOM() {
+    this.handleNavigationClick_ = this.foundation_.handleNavigationClick.bind(this.foundation_);
+    this.handleWindowResize_ = this.foundation_.handleWindowResize.bind(this.foundation_);
+    this.handleTargetScroll_ = this.foundation_.handleTargetScroll.bind(this.foundation_);
+
+    this.scrollTarget_.addEventListener('scroll', this.handleTargetScroll_ as EventListener);
+
+    if (this.navIcon_) {
+      this.navIcon_.addEventListener('click', this.handleNavigationClick_ as EventListener);
+    }
+
+    const isFixed = this.root_.classList.contains(cssClasses.FIXED_CLASS);
+    const isShort = this.root_.classList.contains(cssClasses.SHORT_CLASS);
+    if (!isShort && !isFixed) {
+      window.addEventListener('resize', this.handleWindowResize_ as EventListener);
+    }
+  }
+
   destroy() {
     this.iconRipples_.forEach((iconRipple) => iconRipple.destroy());
+    this.scrollTarget_.removeEventListener('scroll', this.handleTargetScroll_ as EventListener);
+    if (this.navIcon_) {
+      this.navIcon_.removeEventListener('click', this.handleNavigationClick_ as EventListener);
+    }
+    const isFixed = this.root_.classList.contains(cssClasses.FIXED_CLASS);
+    const isShort = this.root_.classList.contains(cssClasses.SHORT_CLASS);
+    if (!isShort && !isFixed) {
+      window.removeEventListener('resize', this.handleWindowResize_ as EventListener);
+    }
     super.destroy();
   }
 
   setScrollTarget(target: EventTarget) {
     // Remove scroll handler from the previous scroll target
-    this.foundation_.destroyScrollHandler();
+    if (this.handleTargetScroll_) {
+      this.scrollTarget_.removeEventListener('scroll', this.handleTargetScroll_ as EventListener);
```

Comment:
`this.handleTargetScroll_` is non-nullable so we don't need this check.

Same below.